### PR TITLE
Use standard directories when running on Linux

### PR DESCRIPTION
**Pull request by agersant**
Proposed fix for https://github.com/agersant/polaris/issues/39

### CODE REVIEW

File: res/unix/Makefile [hunk 1]
```diff
@@ -1,22 +1,42 @@
-POLARIS_BIN_DIR := ~/.local/bin/polaris
-POLARIS_DATA_DIR := ~/.local/share/polaris
+.PHONY: all build install install-bin install-data clean uninstall
+
+BINDIR ?= /usr/local/bin
+DATADIR ?= /usr/local/share
```
Comment (fgaz) [inline res/unix/Makefile:4]:
```suggestion
PREFIX ?= /usr/local
BINDIR ?= $(PREFIX)/bin
DATADIR ?= $(PREFIX)/share
```

File: res/unix/Makefile [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
 uninstall:
-	rm -r $(POLARIS_BIN_DIR)
-	rm -r $(POLARIS_DATA_DIR)
+	rm $(POLARIS_BIN_PATH)
+	rm -r $(POLARIS_WEB_DIR)
+	rm -r $(POLARIS_SWAGGER_DIR)
+	rm -r $(POLARIS_DB_DIR)
+	rm -r $(POLARIS_LOG_DIR)
+	rm -r $(POLARIS_CACHE_DIR)
+	rm -r $(POLARIS_PID_DIR)
```
Comment (fgaz) [inline res/unix/Makefile:42]: I think data files shouldn't be removed on uninstall

**Review by fgaz — COMMENTED** [review]
Just a couple suggestions for now.

i'm still testing this, but I think there'll also be some regressions:

* Not being able to run the installed program as a normal user, since as far as I can see there's no way to override the data/cache/log directory
* Not being able to run the tests after running `make` (not that important)

File: res/unix/Makefile (hunk 1 above)
Comment (agersant) [reply res/unix/Makefile]: I just pushed some follow-up changes. Your suggestion lead me to following the [Make manual](https://www.gnu.org/software/make/manual/html_node/Directory-Variables.html#Directory-Variables) more closely. The data/cache/log directories are now also under `/usr/local` by default, but can be overridden by setting `LOCALSTATEDIR`.

Tests should work regardless of all this as they do not rely on any of these values.

File: res/unix/Makefile (hunk 2 above)
Comment (agersant) [reply res/unix/Makefile:47]: I can't find any official recommendation on this, but I think as a user I would prefer everything gone from my system. I will leave this as-is unless there is a strong argument for leaving files behind.

**Comment by agersant** [comment]
I still want to give this a quick test in a Linux VM myself before merging it, but is it all-clear on your end @fgaz?

Also cc @ogarcia as this will probably have impact on polaris-docker.

**Comment by fgaz** [comment]
Sorry, I've been a bit busy. I'll give this some proper testing next week

**Comment by agersant** [comment]
No worries!

**Comment by fgaz** [comment]
> next week

Or next month ;)

File: res/unix/Makefile [hunk 3]
```diff
@@ -1,22 +1,47 @@
...
+install: install-bin install-data
+
+install-bin: build
+	install -Dm755 ./target/release/polaris $(POLARIS_BIN_PATH)
+
+install-data:
+	install -d $(POLARIS_WEB_DIR)
+	install -d $(POLARIS_SWAGGER_DIR)
+	cp -r ./web $(POLARIS_WEB_DIR)
+	cp -r ./swagger $(POLARIS_SWAGGER_DIR)
```
Comment (fgaz) [inline res/unix/Makefile:35]: This makes a redundant web/web directory. The `-T` flag fixes this:
```suggestion
	cp -rT ./web $(POLARIS_WEB_DIR)
	cp -rT ./swagger $(POLARIS_SWAGGER_DIR)
```

File: src/main.rs [hunk 4]
```diff
@@ -113,6 +128,14 @@ fn main() -> Result<()> {
...
+		"",
+		"cache",
+		"set the directory to use as cache",
+		"DIRECTORY",
+	);
+	options.optopt("", "log", "set the path to the log file", "FILE");
+	options.optopt("", "pid", "set the path to the pid file", "FILE");
 	options.optopt(
 		"l",
 		"log",
```
Comment (fgaz) [inline src/main.rs:141]: Conflicting options

Comment (agersant) [reply src/main.rs:141]: Good catch

**Comment by fgaz** [comment]
I just tested again, and found a problem with the `-system`/`-xdg` split. If I do:

`make PREFIX=whatever`
`make install-system PREFIX=whatever` (or `install-xdg`)

then the default data paths are going to be relative to the current directory. This is because the installed executable is the one that as built with `make all` (-> `make build`), which does not set the variables.

**Comment by agersant** [comment]
Wow, I would 100% have missed that and shipped it. Thanks again for your help with this work.

I fixed this issue by splitting the `build` command into `build-system` and `build-xdg`, defaulting to `build-system` when running through `make all` or `make build`.

**Comment by agersant** [comment]
@fgaz Would you have time to give this another try before I merge it for good?

**Comment by fgaz** [comment]
@agersant I just did a quick test, and it seems to work great!

**Pull request merged**